### DISCUSSION TITLE
Object literal spread - Compatibility issue with Edge 12 - 18

### DISCUSSION THREAD

**Discussion by bluedge** [twbs/bootstrap#31682] [General] [answered]
I noticed the usage of Object Literal spread in the JS code - It is not compatible with Edge 12 to 18 and maybe other browsers. This is not possible to polypill it either (as far as I know).

Example:
```
const attributes = { ...element.dataset };
const attributes = Object.assign({}, element.dataset);
```

Maybe it could replaced by Object.assign() for a better compatibility or Transpiled. Is that something you plan to do in future releases?

**Comment by Johann-S** [comment, marked as answer]
Hi @bluedge,

The compatibility is handle by Babel and we run our unit tests on Edge so don't worry it works 👍

**Reply by bluedge** [reply]
Hi @Johann-S 
Thanks for answering this. Here is the test on Window 10, Edge 18. The JS file is the one compiled from `npm run build`  It looks like babel hasn't worked its magic on this one.

Console - 2 screenshot
https://imgur.com/a/AsgckL5

**Reply by coliff** [reply]
FYI: I think support for Legacy Edge (18 and lower) will be dropped for v5 with #30986

**Reply by bluedge** [reply]
Maybe it could be an option to have in the settings. To support or not earlier Edge versions.
IMHO, as much as I hate any version of IE and early Edge browsers, Internet is about reaching to a maximum of people regardless of their ability to afford the latest computers.

**Reply by coliff** [reply]
I agree generally about reaching a wide audience and having wide browser support where possible, but Microsoft are automatically pushing updates to Windows to remove Legacy Edge and replace it with new Chromium Edge so we should see Legacy Edge marketshare rapidly - and Bootstrap v5 won't hit final release for a good few months yet so by the time its ready it'll be less than 0.05% ([just as Edge 17 is now](https://caniuse.com/usage-table)).

**Reply by bluedge** [reply]
You are right by the time the V5 is out Edge might live in a museum. It always amaze me how IE 11 is still higher than 1%
